File: 프로그래머스/2/12946. 하노이의 탑/하노이의 탑.java
import java.util.*;
import java.util.stream.*;
class Solution {
    List<int[]> result = new ArrayList();
    public int[][] solution(int n) {
        move(1, 3, 2, n);
        return result.stream().toArray(int[][]::new);
    }
    
    public void move(int start, int end, int other, int cnt){
        if (cnt == 1){
            result.add(new int[] { start, end });
            return ;
        }
        
        move(start, other, end, cnt - 1);
        move(start, end, other, 1);
        move(other, end, start, cnt - 1);
    }
}

File: 프로그래머스/2/12946. 하노이의 탑/하노이의 탑.py
def solution(n):
    answer = []
    def recur(s, e, other, n):
        if n == 1:
            answer.append([s, e])
            return
        
        recur(s, other, e, n - 1)
        recur(s, e, other, 1)
        recur(other, e, s, n - 1)
        
    recur(1, 3, 2, n)
    
    return answer
        
    